src/Field/NominatimSearch/NominatimSearch.jsx

```diff
@@ -0,0 +1,276 @@
+
+    /**
+     * Limit search results to a specific country (or a list of countries).
+     * <countrycode> should be the ISO 3166-1alpha2 code, e.g. gb for the United
+     * Kingdom, de for Germany, etc.
+     * @type {String}
+     */
+    countrycodes: 'de',
+
+    /**
+     * The minimal amount of characters entered in the input to start a search.
+     * @type {Number}
+     */
+    minChars: 3,
+
+    /**
+     * The style object passed to the AutoComplete.
+     * @type {Object}
+     */
+    style: {
+      width: 200
+    }
+  }
+
+  /**
+   * Create the NominatimSearch.
+   *
+   * @param {Object} props The initial props.
+   * @constructs NominatimSearch
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchTerm: '',
+      dataSource: []
+    };
+    this.onUpdateInput = this.onUpdateInput.bind(this);
+    this.onMenuItemSelected = this.onMenuItemSelected.bind(this);
+  }
+
+  /**
+   * Called if the input of the AutoComplete is being updated. It sets the
+   * current inputValue as searchTerm and starts a search if the inputValue has
+   * a length of at least `this.props.minChars` (default 3).
+   *
+   * @param {String|undefined} inputValue The inputValue. Undefined if clear btn
+   *                                      is pressed.
+   */
+  onUpdateInput(inputValue) {
+    this.setState({
+      dataSource: []
+    });
+
+    this.setState({
+      searchTerm: inputValue || ''
+    }, () => {
+      if (this.state.searchTerm.length >= this.props.minChars) {
+        this.doSearch();
+      }
+    });
+  }
+
+  /**
+   * Perform the search.
+   */
+  doSearch() {
+    const baseParams = {
+      format: this.props.format,
+      viewbox: this.props.viewbox,
+      bounded: this.props.bounded,
+      polygon_geojson: this.props.polygon_geojson,
+      addressdetails: this.props.addressdetails,
+      limit: this.props.limit,
+      countrycodes: this.props.countrycodes,
+      q: this.state.searchTerm
+    };
+
+    const getRequestParams = UrlUtil.objectToRequestString(baseParams);
+
+    fetch(`${this.props.nominatimBaseUrl}${getRequestParams}`)
+      .then(response => response.json())
+      .then(this.onFetchSuccess.bind(this))
+      .catch(this.onFetchError.bind(this));
+  }
+
+  /**
+   * This function gets called on success of the nominatim fetch.
+   * It sets the response as dataSource.
+   *
+   * @param {Array<object>} response The found features.
+   */
+  onFetchSuccess(response) {
+    this.setState({
+      dataSource: response
+    });
+  }
+
+  /**
+   * This function gets called when the nomintim fetch returns an error.
+   * It logs the error to the console.
+   *
+   * @param {String} error The errorstring.
+   */
+  onFetchError(error) {
+    Logger.error(`Error while requesting Nominatim: ${error}`);
+  }
+
+  /**
+   * The function describes what to do when an item is selected.
+   *
+   * @param {value} key The key of the selected option.
+   */
+  onMenuItemSelected(key) {
```

Comment:
We should introduce a `onSelect` prop to the component, so applications may override this behaviour (e.g. to set the map view via dispatching an action or similiar).

Reply:
Ah, never mind. Just noticed that the `AutoComplete` compoent has an `onSelect` that can be used already :duck: